src/components/manage-registry/manage-agents/ManageAgents.tsx

```diff
@@ -21,24 +21,41 @@ import {
 import { useWalletInterface } from "@/services/wallets/useWalletInterface";
 import { WalletInterface } from "@/services/wallets/walletInterface";
 import { useTokenIdentityRegistryAgents } from "@/hooks/useTokenIdentityRegistryAgents";
+import { ethers } from "ethers";
 
 type ManageAgentsProps = {
   setUpdateTxResult: (res?: string) => void;
   setUpdateTxError: (err?: string) => void;
   onClose: () => void;
   registry?: EvmAddress;
+  identityItems: IdentityItem[];
 };
 
 export function ManageAgents({
   setUpdateTxResult,
   setUpdateTxError,
   registry,
+  identityItems,
   onClose,
 }: ManageAgentsProps) {
+  const [selectedAgentToAdd, setSelectedAgentToAdd] = useState<EvmAddress>();
   const [selectedAgent, setSelectedAgent] = useState<EvmAddress>();
-  const [newUserAgentAddress, setNewUserAgentAddress] = useState<string>();
-  const { walletInterface } = useWalletInterface();
-  const { filteredAgents } = useTokenIdentityRegistryAgents(registry);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [agentManualError, setAgentManualError] = useState<string>();
+  const [agentSelectError, setAgentSelectError] = useState<string>();
```

Comment:
we can have default `""` value and drop the additional typing and simplify from `string | undefined` to just `string` automatically